packages/@aws-cdk/aws-stepfunctions-tasks/lib/batch/submit-job.ts

```diff
@@ -325,3 +332,7 @@ export class BatchSubmitJob extends sfn.TaskStateBase {
     };
   }
 }
+
+function tokenOrString(value: any) {
+  return Token.isUnresolved(value) ? value : value.toString();
```

Comment:
@rix0rrr, I'm a bit confused about the necessity for something like this. Originally we had 
```ts
Value: `${maybeToken}`
```
But this does not work when `maybeToken` is a token. Not sure why this is the case. I looked for prior art for something like this and couldn't find anything. The typical `Lazy.number({ produce: () => {}})` isn't applicable because the token is a `JsonPath` (and there's some fancy footwork that goes on behind the scenes for JsonPaths).